src/raw/mod.rs

```diff
@@ -31,43 +35,57 @@ mod requests;
 ///
 /// ```rust
 /// # use tikv_client::ColumnFamily;
+/// # use std::convert::TryFrom;
 ///
-/// let cf = ColumnFamily::from("write");
-/// let cf = ColumnFamily::from(String::from("write"));
-/// let cf = ColumnFamily::from(&String::from("write"));
+/// let cf = ColumnFamily::try_from("write").unwrap();
+/// let cf = ColumnFamily::try_from(String::from("write")).unwrap();
 /// ```
 ///
 /// **But, you should not need to worry about all this:** Many functions which accept a
 /// `ColumnFamily` accept an `Into<ColumnFamily>`, which means all of the above types can be passed
 /// directly to those functions.
-#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+#[derive(Clone, Eq, PartialEq, Hash, Debug)]
 pub enum ColumnFamily {
     Default,
     Lock,
     Write,
-    Version,
+    VersionDefault,
 }
 
-impl<T: Into<String>> From<T> for ColumnFamily {
-    fn from(i: T) -> ColumnFamily {
-        match i.into().as_str() {
-            "default" => ColumnFamily::Default,
-            "lock" => ColumnFamily::Lock,
-            "write" => ColumnFamily::Write,
-            "versioncf" => ColumnFamily::Version,
-            _ => unreachable!(),
+impl TryFrom<&str> for ColumnFamily {
+    type Error = Error;
+
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        match value {
+            "default" => Ok(ColumnFamily::Default),
+            "lock" => Ok(ColumnFamily::Lock),
+            "write" => Ok(ColumnFamily::Write),
+            "ver_default" => Ok(ColumnFamily::VersionDefault),
+            s => Err(ColumnFamilyError(s.to_owned()).into()),
         }
     }
 }
 
-impl ToString for ColumnFamily {
-    fn to_string(&self) -> String {
-        match self {
-            ColumnFamily::Default => "default".to_owned(),
-            ColumnFamily::Lock => "lock".to_owned(),
-            ColumnFamily::Write => "write".to_owned(),
-            ColumnFamily::Version => "versioncf".to_owned(),
-        }
+impl TryFrom<String> for ColumnFamily {
+    type Error = Error;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        TryFrom::try_from(value.as_str())
+    }
+}
+
+impl fmt::Display for ColumnFamily {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "{}",
+            match self {
+                ColumnFamily::Default => "default",
+                ColumnFamily::Lock => "lock",
+                ColumnFamily::Write => "write",
+                ColumnFamily::VersionDefault => "ver_default",
+            }
+        )
```

Comment:
style nit: I would put the `match` on the 'outside' rather than inside the `write!`, you can use `f.write_str("foo")` for each case, which is marginally simpler than using `write!`